src/keylessBackup/saga.ts

```diff
@@ -110,8 +119,49 @@ function* handleKeylessBackupRestore({
     encryptionAddress,
   })
 
-  // TODO(ACT-778): use mnemonic
-  yield* call(decryptPassphrase, torusKeyshareBuffer, valoraKeyshareBuffer, encryptedMnemonic)
+  const decryptedMnemonic = yield* call(
+    decryptPassphrase,
+    torusKeyshareBuffer,
+    valoraKeyshareBuffer,
+    encryptedMnemonic
+  )
+
+  const { privateKey } = yield* call(generateKeysFromMnemonic, decryptedMnemonic)
+  if (!privateKey) {
+    throw new Error('Failed to convert mnemonic to hex')
+  }
+  const backupAccount = privateKeyToAddress(privateKey)
+  if (!(yield* call(walletHasBalance, backupAccount))) {
+    // show zero balance modal
+    yield* put(keylessBackupShowZeroBalance())
+
+    // wait for user to click continue or bail from keyless backup
+    const [_continueAction, bailAction] = yield* race([
+      take(keylessBackupAcceptZeroBalance.type),
+      take(keylessBackupBail.type),
+    ])
+    if (bailAction) {
+      // TODO(ACT-874): Navigate to ImportSelect screen
+      return
+    }
+  }
+
+  const account: string | null = yield* call(
+    assignAccountFromPrivateKey,
+    privateKey,
+    decryptedMnemonic
+  )
+  if (!account) {
+    throw new Error('Failed to assign account from private key')
+  }
+  // Set key in phone's secure store
+  yield* call(storeMnemonic, decryptedMnemonic, account)
+  // Set backup complete so user isn't prompted to do backup flow
+  yield* put(setBackupCompleted())
```

Comment:
I guess we have two "backup completed" states now, one for the quiz and one for the cloud backup. We check both states before showing the backup flow. So is this required since we set cloud backup as completed below? If we do want to set this, then is this something that should be set when cloud backup is setup as well?

Reply:
Ah, yea we check both. I don't think we should to / need to set the regular backup completed here. thanks!